core/validation/pre_transformation_validate.py

```diff
@@ -0,0 +1,168 @@
+        col, row = expected_values
+        fund_type_of_entered_place = PLACE_TO_FUND_TYPE.get(str(workbook[sheet].iloc[col][row]).strip())
+        expected_values = tuple(map(fund_type_acronym_to_full_name.get, fund_type_of_entered_place))
+
+        conflicting_input_checks.append(
+            PreTransformationCheck(sheet=sheet, column=column, row=row, expected_values=expected_values, type=type)
+        )
+
+    failures = check_values(workbook, conflicting_input_checks, "wrong_input_failure")
+
+    if failures:
+        raise ValidationError(validation_failures=failures)
+
+
+def check_values(
+    workbook: dict,
+    failure_list: list,
+    failure_type: str,
+) -> list[uf.WrongInputFailure] | list[uf.UnauthorisedSubmissionFailure] | None:
+    """Checks values in the workbook against the expected values and returns failures.
+
+    :param workbook: A dictionary where keys are sheet names and values are pandas
+                     DataFrames representing each sheet in the submission.
+    :param failure_list: A list of tuples specifying checks to be performed.
+    :param failure_type: Type of failure to be checked, e.g., "wrong_input_failure".
+    :return: A list of validation failures (instances of FailureClass) if any, else None.
+    """
+
+    str_repr_of_failure_to_actual_failure = {
+        "wrong_input_failure": uf.WrongInputFailure,
+        "unauthorised_submission_failure": uf.UnauthorisedSubmissionFailure,
+    }
+    failure = str_repr_of_failure_to_actual_failure[failure_type]
+    failures = []
+
+    for sheet, column, row, expected_values, type in failure_list:
+        if workbook[sheet].iloc[column][row] not in expected_values:
+            failures.append(
+                failure(
+                    value_descriptor=type,
+                    entered_value=workbook[sheet].iloc[column][row],
+                    expected_values=expected_values,
+                )
+            )
+
+    return failures
+
+
+def validate_sign_off(workbook: dict[str, pd.DataFrame]) -> list[uf.GenericFailure] | None:
+    """Validates Name, Role, and Date for the Review & Sign-Off Section
+
+    :param workbook: A dictionary where keys are sheet names and values are pandas
+                     DataFrames representing each sheet in the Round 4 submission.
+    :return: ValidationErrors
+    """
+    sheet = workbook["8 - Review & Sign-Off"]
+    sheet.replace(r"", np.nan, inplace=True)
+
+    section_151_text_cells = [6, 7, 9]
+    town_board_chair_cells = [13, 14, 16]
+
+    failures = []
+
+    for y_axis in [*town_board_chair_cells, *section_151_text_cells]:
+        if pd.isnull(sheet.iloc[y_axis, 2]):
+            failures.append(
+                uf.GenericFailure(
+                    sheet="Review & Sign-Off", section="-", cell_index="C" + str(y_axis + 2), message=msgs.BLANK
+                )
+            )
+
+    return failures
```

Comment:
Perhaps this should go somewhere else now? Probably best here `core.validation.specific_validations.towns_fund_round_four`